File: DELETE BEFORE PRESENTATION/git-cleanup/remove_git_history.py
"""
GRC Risk Register — Safe Presentation Git Removal Tool
=====================================================
Safely removes the .git/ directory from a specified target copy of the project.

Safety Invariants:
1. NEVER operates on the active working tree silently.
2. Requires explicit target directory specification and user confirmation.
3. Deletes ONLY the .git/ folder.
4. Preserves all application source code, databases, templates, and documentation.
"""

import os
import sys
import shutil
import stat
import argparse

def remove_readonly(func, path, excinfo):
    """Clear the read-only bit and retry removal."""
    try:
        os.chmod(path, stat.S_IWRITE)
        func(path)
    except Exception:
        pass

def remove_git_history(target_dir, force=False):
    target_path = os.path.abspath(target_dir)

    print("=======================================================")
    print("  GRC Risk Register — Presentation Git Removal Tool")
    print("=======================================================")
    print(f"\nTarget directory: {target_path}")

    if not os.path.isdir(target_path):
        print(f"[ERROR] Target directory does not exist: {target_path}")
        return False

    git_dir = os.path.join(target_path, ".git")

    if not os.path.exists(git_dir):
        print(f"[INFO] No .git directory found at: {git_dir}")
        print("[INFO] Nothing to remove.")
        return True

    print(f"[FOUND] Git version history found at: {git_dir}")
    print("\nWARNING: This will permanently delete the Git history from THIS target copy.")
    print("Source files, templates, and database will remain completely untouched.")
    print("=======================================================")

    if not force:
        try:
            confirm = input(f"\nAre you sure you want to permanently remove Git history from THIS COPY? (Y/N): ").strip().upper()
        except (KeyboardInterrupt, EOFError):
            print("\nOperation cancelled.")
            return False

        if confirm not in ("Y", "YES"):
            print("[CANCELLED] Operation cancelled by user. No changes made.")
            return False

    print("\n[REMOVING] Deleting .git directory...")
    try:
        # Clear attributes and remove tree
        for root, dirs, files in os.walk(git_dir):
            for f in files:
                p = os.path.join(root, f)
                try: os.chmod(p, stat.S_IWRITE)
                except Exception: pass
            for d in dirs:
                p = os.path.join(root, d)
                try: os.chmod(p, stat.S_IWRITE)
                except Exception: pass

        shutil.rmtree(git_dir, onerror=remove_readonly)

        if not os.path.exists(git_dir):
            print(f"[SUCCESS] Git history (.git/) removed successfully from:\n  {target_path}")
            return True
        else:
            print(f"[ERROR] Failed to completely remove .git directory.")
            return False
    except Exception as e:
        print(f"[ERROR] Exception during removal: {e}")
        return False

def main():
    parser = argparse.ArgumentParser(description="Safely remove .git history from a target presentation copy.")
    parser.add_argument("target", nargs="?", default="", help="Path to target directory to clean")
    parser.add_argument("-y", "--yes", action="store_true", help="Skip confirmation prompt")
    args = parser.parse_args()

    target = args.target.strip()
    if not target:
        print("=======================================================")
        print("  GRC Risk Register — Presentation Git Removal Tool")
        print("=======================================================")
        print("\nPlease enter the path to the presentation copy folder you wish to clean.")
        print("Example: C:\\Users\\Name\\Desktop\\Presentation Copy\n")
        try:
            target = input("Enter target folder path: ").strip()
        except (KeyboardInterrupt, EOFError):
            print("\nOperation cancelled.")
            sys.exit(1)

        if not target:
            print("[ERROR] No target folder specified. Refusing to operate on unknown directory.")
            sys.exit(1)

    success = remove_git_history(target, force=args.yes)
    sys.exit(0 if success else 1)

if __name__ == "__main__":
    main()

File: DELETE BEFORE PRESENTATION/git-cleanup/Remove_Git_History.cs
using System;
using System.IO;

namespace GRCRiskRegister.GitCleanup
{
    class Program
    {
        static int Main(string[] args)
        {
            Console.Title = "GRC Risk Register — Remove Git History";
            Console.WriteLine("=======================================================");
            Console.WriteLine("  GRC Risk Register — Git History Removal Tool");
            Console.WriteLine("=======================================================");

            string baseDir = AppDomain.CurrentDomain.BaseDirectory;
            string projectRoot = FindProjectRoot(baseDir);

            if (string.IsNullOrEmpty(projectRoot))
            {
                Console.ForegroundColor = ConsoleColor.Red;
                Console.WriteLine("\n[ABORTED] Could not automatically locate the GRC project root.");
                Console.WriteLine("Safety check failed: 'Grc Risk Management Code' folder not found in parent hierarchy.");
                Console.WriteLine("Refusing to operate on unknown directory.\n");
                Console.ResetColor();
                return 1;
            }

            Console.WriteLine("\n[INFO] Target Project Root: " + projectRoot);
            Console.WriteLine("\n[REMOVING] Deleting all Git metadata and repository history...");

            int removedCount = 0;
            int failedCount = 0;

            // 1. Delete all .git directories recursively inside projectRoot
            try
            {
                string[] allDirs = Directory.GetDirectories(projectRoot, "*", SearchOption.AllDirectories);
                foreach (string d in allDirs)
                {
                    if (string.Equals(Path.GetFileName(d), ".git", StringComparison.OrdinalIgnoreCase))
                    {
                        Console.WriteLine("[REMOVING] Deleting folder: " + d);
                        if (DeleteDirectorySafe(d)) removedCount++;
                        else failedCount++;
                    }
                }
            }
            catch { }

            // Check root .git directly
            string rootGit = Path.Combine(projectRoot, ".git");
            if (Directory.Exists(rootGit))
            {
                Console.WriteLine("[REMOVING] Deleting folder: " + rootGit);
                if (DeleteDirectorySafe(rootGit)) removedCount++;
                else failedCount++;
            }

            // 2. Delete Git configuration files (.gitignore, .gitattributes, .gitmodules)
            string[] gitFiles = new string[] { ".gitignore", ".gitattributes", ".gitmodules" };
            foreach (string gf in gitFiles)
            {
                try
                {
                    string[] found = Directory.GetFiles(projectRoot, gf, SearchOption.AllDirectories);
                    foreach (string f in found)
                    {
                        Console.WriteLine("[REMOVING] Deleting file: " + f);
                        try
                        {
                            File.SetAttributes(f, FileAttributes.Normal);
                            File.Delete(f);
                            removedCount++;
                        }
                        catch (Exception ex)
                        {
                            Console.WriteLine("[ERROR] Could not delete " + f + ": " + ex.Message);
                            failedCount++;
                        }
                    }
                }
                catch { }
            }

            // 3. Post-cleanup audit
            bool clean = !Directory.Exists(rootGit);
            Console.WriteLine("\n=======================================================");
            Console.WriteLine("  CLEANUP SUMMARY");
            Console.WriteLine("=======================================================");
            Console.WriteLine("Items removed: " + removedCount);
            Console.WriteLine("Items failed:  " + failedCount);

            if (clean && failedCount == 0)
            {
                Console.ForegroundColor = ConsoleColor.Green;
                Console.WriteLine("\n[SUCCESS] Project is 100% clean of all Git metadata and version control history.");
                Console.WriteLine("Source code, presentation database, and application runtime are intact.\n");
                Console.ResetColor();
            }
            else
            {
                Console.ForegroundColor = ConsoleColor.Yellow;
                Console.WriteLine("\n[WARNING] Some Git items could not be completely removed.\n");
                Console.ResetColor();
            }

            return clean ? 0 : 1;
        }

        private static string FindProjectRoot(string startDir)
        {
            DirectoryInfo current = new DirectoryInfo(startDir);
            for (int i = 0; i < 6 && current != null; i++)
            {
                string appCode = Path.Combine(current.FullName, "Grc Risk Management Code");
                string launcher = Path.Combine(current.FullName, "Start GRC Risk Register.exe");
                string readme = Path.Combine(current.FullName, "README.md");

                if (Directory.Exists(appCode) && (File.Exists(launcher) || File.Exists(readme)))
                {
                    return current.FullName;
                }

                current = current.Parent;
            }
            return null;
        }

        private static bool DeleteDirectorySafe(string path)
        {
            try
            {
                if (!Directory.Exists(path)) return true;
                foreach (string f in Directory.GetFiles(path, "*", SearchOption.AllDirectories))
                {
                    try { File.SetAttributes(f, FileAttributes.Normal); } catch { }
                }
                Directory.Delete(path, true);
                return !Directory.Exists(path);
            }
            catch
            {
                return false;
            }
        }
    }
}
